File: 242 Valid Anagram/242 Valid Anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        return { x: s.count(x) for x in set(s) } == { x: t.count(x) for x in set(t) }



class Solution(object):
    def isAnagram(self, s: str, t: str) -> bool:
        if (len(s) != len(t)): 
            return False
        temp = [0] * 26
        a = ord('a')
        for i, value in enumerate(list(s)):
            temp[ord(value) - a] += 1
            temp[ord(t[i]) - a] -= 1
        for j in temp:
            if (j != 0):
                return False
        return True


class Solution:
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        for char in set(s):
            if s.count(char) != t.count(char):
                return False
        return True

File: 242 Valid Anagram/242 Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int sL = s.length(), tL = t.length();
        if (sL != tL) return false;
        HashMap<Character, Integer> temp = new HashMap<>();
        for (int i = 0; i < sL; ++i) {
            temp.put(s.charAt(i), temp.getOrDefault(s.charAt(i), 0) + 1);
            temp.put(t.charAt(i), temp.getOrDefault(t.charAt(i), 0) - 1);
        }
        for (Integer val : temp.values()) {
            if (val != 0) return false;
        }
        return true;
    }

    
    public boolean isAnagram2(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        int[] counter = new int[26];
        for (int i = 0; i < s.length(); i++) {
            counter[s.charAt(i) - 'a']++;
            counter[t.charAt(i) - 'a']--;
        }
        for (int count : counter) {
            if (count != 0) {
                return false;
            }
        }
        return true;
    }

}
